src/test/java/org/folio/rest/api/HoldingsStorageTest.java

```diff
@@ -611,7 +611,19 @@ public void tenantIsRequiredForGettingAllHoldings() throws InterruptedException,
     assertThat(response.getBody(), is("Unable to process request Tenant must be set"));
   }
 
-  @Test 
+  @Test
+  public void optimisticLockingVersion() {
+    UUID holdingId = createInstanceAndHolding(mainLibraryLocationId);
+    JsonObject holding = getById(holdingId.toString()).getJson();
+    holding.put(PERMANENT_LOCATION_ID_KEY, annexLibraryLocationId);
+    // updating with current _version 1 succeeds and increments _version to 2
+    assertThat(update(holding).getStatusCode(), is(204));
+    holding.put(PERMANENT_LOCATION_ID_KEY, secondFloorLocationId);
+    // updating with outdated _version 1 fails, current _version is 2
+    assertThat(update(holding).getStatusCode(), is(409));
```

Comment:
Should the consumer increment it or it is autoincremented if difference is 1?

For example, I've read an item from database, that has `_version: 3`, than I updated item shelving order. Assuming no concurrent updates, should I increment `_version` to `4` or it is incremented by the locking mechanism? 

I'm asking, because there is a migration class `org.folio.services.migration.item.ItemShelvingOrderMigrationService` that does exactly what I described.

Reply:
The consumer/client should never increment it. Only the database trigger created by RMB should increment it to ensure that the check and the increment are in the same transaction to avoid any race condition.

Reply:
Sounds great, thanks!